compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -252,7 +252,7 @@ enum ImplTraitContext<'b, 'a> {
         // FIXME(impl_trait): but `required_region_bounds` will ICE later
         // anyway.
         capturable_lifetimes: &'b mut FxHashSet<hir::LifetimeName>,
-        /// Origin: Either OpaqueTyOrigin::Misc or OpaqueTyOrigin::Binding,
+        /// Origin: Either OpaqueTyOrigin::TyAlias or OpaqueTyOrigin::Misc,
```

Comment:
since you've removed `OpaqueTyOrigin::Misc`:

```suggestion
        /// Origin: OpaqueTyOrigin::TyAlias,
```

and if it's really always an `OpaqueTyOrigin::TyAlias` maybe there's no need to have this `origin` field anymore ?

No big deal either way.